src/main/java/com/sourcegraph/cody/PromptPanel.kt

```diff
@@ -264,12 +251,41 @@ class PromptPanel(project: Project, private val chatSession: ChatSession) : JLay
     val UP = CustomShortcutSet(KEY_UP)
     val DOWN = CustomShortcutSet(KEY_DOWN)
     val TAB = CustomShortcutSet(KEY_TAB)
+
+    private val atExpressionPattern = """(@(?:\\\s|\S)*)(?:\s|$)""".toRegex()
+
+    fun findAtExpressions(text: String): List<AtExpression> {
+      val matches = atExpressionPattern.findAll(text)
+      val expressions = ArrayList<AtExpression>()
+      for (match in matches) {
+        val mainMatch = match.groups[0] ?: continue
+        val prevIndex = mainMatch.range.first - 1
+        // filter out things like email addresses
+        if (prevIndex >= 0 && !text[prevIndex].isWhitespace()) continue
+
+        val subMatch = match.groups[1]
+        if (subMatch != null) {
+          val value = subMatch.value.substring(1).replace("\\ ", " ")
+          expressions.add(
+              AtExpression(subMatch.range.first, subMatch.range.last + 1, subMatch.value, value))
+        }
+      }
+      return expressions
+    }
+
+    private fun findContextFiles(contextItems: List<ContextItem>, text: String): List<ContextItem> {
+      val atExpressions = findAtExpressions(text)
+      return contextItems.filter { f -> atExpressions.any { it.value == f.displayPath() } }
+    }
   }
 }
 
 data class DisplayedContextFile(val contextItem: ContextItem) {
   override fun toString(): String {
-    return displayPath(contextItem)
+    val contextItemFile = contextItem as? ContextItemFile
+    val isTooLarge = contextItemFile?.title == "large-file" || contextItemFile?.isTooLarge == true
+    val warnIfNeeded = if (isTooLarge) "<i> - ⚠ File too large</i>" else ""
```

Comment:
Can we bundle this text?

Reply:
I'm doing a followup to include a file ranges, I will fix there if you don't mind.